src/Expression/AbstractExpressionBuilder.php

```
@@ -46,6 +47,10 @@ public function __construct(private QueryBuilderInterface $queryBuilder)
      */
     public function build(ExpressionInterface $expression, array &$params = []): string
     {
+        if (!$expression instanceof Expression) {
+            throw new InvalidArgumentException(static::class . ' could only be used with Expression.');
```

Should we throw `InvalidArgumentException` or `InvalidConfigException` here?